schema/xmlcatalog.xml

```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0"?>
+<catalog xmlns="urn:oasis:names:tc:entity:xmlns:xml:catalog">
```

Comment:
Short answer: we do not need DTD when we already use XSD schema files.
But I am not an XML expert, I put below the result of a question to chatGPT about whether a DTD should be included in this particular XML catalog for CycloneDX or not.

Is this answer is satisfactory?

***

A Document Type Definition (DTD) is used to define the structure and the legal elements and attributes of an XML document. While DTDs can be useful for ensuring the correctness and validity of an XML document, they are not strictly necessary for all XML documents, especially if you are using XML Schema Definitions (XSD) instead.

In the context of an XML catalog, which is primarily used to map public identifiers and system identifiers to local resources, a DTD is not typically required. XML catalogs are more about resolving URIs to local files rather than enforcing a document structure.

However, if you want to ensure that your XML catalog adheres to a specific structure and follows the XML Catalog standard, you can refer to the DTD for XML catalogs. The XML Catalog specification itself defines a DTD that can be used.

Here’s an example of how you can reference the XML Catalog DTD in your XML catalog file:

```xml
<?xml version="1.0"?>
<!DOCTYPE catalog PUBLIC "-//OASIS//DTD XML Catalogs V1.1//EN" "http://www.oasis-open.org/committees/entity/release/1.1/catalog.dtd">
<!--
  This XML catalog provides mappings for CycloneDX schemas.
  The catalog maps schema URLs to local XSD files to facilitate schema validation without needing internet access.
  Namespace: urn:oasis:names:tc:entity:xmlns:xml:catalog
-->
<catalog xmlns="urn:oasis:names:tc:entity:xmlns:xml:catalog">

  <!-- SPDX Schema -->
  <uri name="http://cyclonedx.org/schema/spdx" uri="spdx.xsd"/>

  <!-- BOM Schemas -->
  <uri name="http://cyclonedx.org/schema/bom/1.0" uri="bom-1.0.xsd"/>
  <uri name="http://cyclonedx.org/schema/bom/1.1" uri="bom-1.1.xsd"/>
  <uri name="http://cyclonedx.org/schema/bom/1.2" uri="bom-1.2.xsd"/>
  <uri name="http://cyclonedx.org/schema/bom/1.3" uri="bom-1.3.xsd"/>
  <uri name="http://cyclonedx.org/schema/bom/1.4" uri="bom-1.4.xsd"/>
  <uri name="http://cyclonedx.org/schema/bom/1.5" uri="bom-1.5.xsd"/>
  <uri name="http://cyclonedx.org/schema/bom/1.6" uri="bom-1.6.xsd"/>

  <!-- Placeholder for future schemas -->
  <!-- <uri name="http://cyclonedx.org/schema/bom/1.x" uri="bom-1.x.xsd"/> -->

</catalog>
```

### Explanation

1. **DOCTYPE Declaration**:
   - The `<!DOCTYPE catalog PUBLIC "-//OASIS//DTD XML Catalogs V1.1//EN" "http://www.oasis-open.org/committees/entity/release/1.1/catalog.dtd">` declaration specifies that this document adheres to the XML Catalog DTD.

2. **Comments**:
   - Comments explain the purpose of the XML catalog and the schema mappings.

### When to Use DTD

Use a DTD if:
- You need to enforce a specific structure in your XML documents.
- You want to validate the XML against the DTD to catch structural errors.

For XML catalogs specifically, adhering to the XML Catalog DTD can help ensure that your catalog files are correct and compliant with the standard.

### When Not to Use DTD

If you are already using XSDs to validate your XML documents and your toolchain supports XSD validation well, you might not need a DTD. XSDs offer more powerful validation features compared to DTDs.

### Summary

While you don't strictly need a DTD for an XML catalog, referencing the XML Catalog DTD can help ensure your catalog file is well-formed and adheres to the standard. This can be particularly useful if the catalog will be used in environments that validate against the DTD. If your use case benefits from this validation, including the DTD reference is a good practice.

Reply:
see https://github.com/ThalesGroup/cyclonedx-specification/pull/1

Reply:
I merged and added your comment in the XML catalog, thank you. I added a couple of other comments as well.

https://github.com/ThalesGroup/cyclonedx-specification/commit/4631813151b94f76dd8aa6b536216b48e779b048